packages/react/src/components/CodeSnippet/CodeSnippet-story.js

```diff
@@ -1,135 +1,321 @@
 /**
- * Copyright IBM Corp. 2016, 2018
+ * Copyright IBM Corp. 2016, 2021
  *
  * This source code is licensed under the Apache-2.0 license found in the
  * LICENSE file in the root directory of this source tree.
  */
 
 import React from 'react';
 import { action } from '@storybook/addon-actions';
-import { withKnobs, boolean, text, select } from '@storybook/addon-knobs';
-import CodeSnippet from '../CodeSnippet';
+import {
+  withKnobs,
+  boolean,
+  text,
+  select,
+  number,
+} from '@storybook/addon-knobs';
+import CodeSnippet from './CodeSnippet';
 import CodeSnippetSkeleton from './CodeSnippet.Skeleton';
 import mdx from './CodeSnippet.mdx';
 
+const lightPropMessage = (
+  <small style={{ display: 'block', paddingBottom: '1rem' }}>
+    The snippet container should never be the same color as the page background.
+    <br />
+    {'Do not use the '}
+    <CodeSnippet type="inline" light>
+      light
+    </CodeSnippet>
+    {' variant on '}
+    <CodeSnippet type="inline" light>
+      $ui-background
+    </CodeSnippet>
+    {' or '}
+    <CodeSnippet type="inline" light>
+      $ui-02
+    </CodeSnippet>
+    .
+  </small>
+);
+
 export default {
   title: 'Components/CodeSnippet',
   component: CodeSnippet,
-  decorators: [withKnobs],
+  subcomponents: {
+    CodeSnippetSkeleton,
+  },
+  decorators: [
+    withKnobs,
+    (story) => (
+      <div className={lightProp().light ? 'bx--tile' : ''}>
+        {lightProp().light && lightPropMessage}
+        <br />
+        {story()}
+      </div>
+    ),
+  ],
   parameters: {
     docs: {
       page: mdx,
     },
   },
 };
 
-const props = () => ({
-  type: select(
-    'Type (type)',
-    {
-      inline: 'inline',
-      'single line': 'single',
-      'multiple line': 'multi',
-    },
-    'inline'
-  ),
-  disabled: boolean('Disabled (disabled)', false),
-  light: boolean('Light variant (light)', false),
-  feedback: text('Feedback text', 'Copied to clipboard'),
-  showMoreText: text('Text for "show more" button', 'Show more'),
-  showLessText: text('Text for "show less" button', 'Show less'),
-  hideCopyButton: boolean('Hide copy button (hideCopyButton)', false),
+const typeOptions = {
+  inline: { inline: 'inline' },
+  single: { single: 'single' },
```

Comment:
Is there any reason why we're changing all these storybook lines? This is adding quite a bit of bloat to the PR